examples/calculator/features/step_definitions/display_steps.rs

```diff
@@ -4,11 +4,15 @@ use support::env::CalculatorWorld;
 
 #[allow(dead_code)]
 pub fn register_steps(c: &mut CucumberRegistrar<CalculatorWorld>) {
-  Then!(c; "^the display( doesn't)? says? (\\d+)", |_, world: &mut CalculatorWorld, (negate, number): (Option<String>, i32)| {
-    if negate.is_none() {
-      InvokeResponse::check_eq(world.calculator.display_contents(), number)
-    } else {
-      InvokeResponse::check_not_eq(world.calculator.display_contents(), number)
+  Then!(c; "^the (?:next )?result is (-)?(\\d+)$", |_, world: &mut CalculatorWorld, (negate, mut val): (Option<String>, i32)| {
```

Comment:
Writing this example, particularly the hoops jumped through for the negative sign, lead me to believe that there should be an easy coersion from Option<T> to bool. I think that could be done without too much trouble, and would satisfy most usecases with optional capures.